src/App.tsx

```diff
@@ -0,0 +1,128 @@
+});
+
+class App extends React.Component<any, State> {
+    state: State = { ...initial, movies: [] };
+
+    constructor(props: any) {
+        super(props);
+
+        this.handleCreate = this.handleCreate.bind(this);
+        this.handleEdit = this.handleEdit.bind(this);
+        this.handleDelete = this.handleDelete.bind(this);
+        this.handleClose = this.handleClose.bind(this);
+        this.showDelete = this.showDelete.bind(this);
+        this.showEdit = this.showEdit.bind(this);
+        this.showNew = this.showNew.bind(this);
+    }
+
+    componentDidMount() {
+        // pseudo-fetch data
+        this.setState({ movies: MOVIES });
+    }
+
+    handleCreate(data: Movie) {
+        const newMovie: MovieItem = {
+            ...data,
+            id: `${Math.round(Math.random() * 100)}`,
+            imageUrl: 'https://media.istockphoto.com/vectors/cinema-and-movie-time-vector-id640312764'
+        }
+        this.setState({ movies: [ ...this.state.movies, newMovie ], isNew: false, isSuccess: true });
+    }
+
+    handleEdit(data: Movie) {
+        const current = this.state.movies.find(m => m.id === this.state.currentId);
+        const ind = this.state.movies.indexOf(current);
+        const updated = Object.assign({}, current, data);
+        const movies = [...this.state.movies];
+        movies.splice(ind, 1, updated);
+
+        this.setState({ ...initial, movies });
+    }
+
+    handleDelete() {
+        const movies = this.state.movies.filter(m => m.id !== this.state.deleteId);
+        this.setState({ ...initial, movies });
+    }
+
+    handleClose() {
+        this.setState(initial);
+    }
+
+    showDelete(id: string) {
+        this.setState({ showPopup: true, deleteId: id });
+    }
+
+    showEdit(id: string) {
+        const current = this.state.movies.find(m => m.id === id);
+        this.setState({ showPopup: true, current, currentId: id });
+    }
+
+    showNew() {
+        this.setState({ showPopup: true, isNew: true });
+    }
+
+    render() {
+        const { movies, showPopup, current, deleteId, isNew, isSuccess } = this.state;
+
+        return (
+            <>
+                <CreateContext.Provider value={this.showNew}>
+                    <Header />
+                </CreateContext.Provider>
+                <ErrorBoundary>
+                    <Movies movies={movies} onEdit={this.showEdit} onDelete={this.showDelete} />
+                </ErrorBoundary>
+                <Footer>
+                    <LogoLink />
+                </Footer>
+
+                {showPopup && (
+                    <PopUp onClose={this.handleClose}>
+                        {deleteId && <Delete onDelete={this.handleDelete} />}
+                        {current && <MovieContainer movie={current} onSubmit={this.handleEdit} />}
+                        {isNew && <MovieContainer onSubmit={this.handleCreate} />}
+                        {isSuccess && <Success />}
+                    </PopUp>
+                )}
```

Comment:
I think that is better to split logic. For example if `deletedId` and `isNew` will be true, we will get a popup with two components inside. It's unlikely event but we need to play to safe from bugs. So I've recommended make several popups
```suggestion
                <PopUp
                  isOpen={showPopup && deleteMovieId} //`deleteId` of what?
                  onClose={this.handleClose}
                >
                   <Delete onDelete={this.handleDelete} /> //better to call this component `DeleteModal` or `DeleteContainer`
                </PopUp>
                
                <PopUp
                  isOpen={showPopup && currentMovie} //just `current` is not informative
                  onClose={this.handleClose}
                >
                   <MovieContainer movie={currentMovie} onSubmit={this.handleEdit} />
                </PopUp>
                
                <PopUp
                  isOpen={showPopup && isNew}
                  onClose={this.handleClose}
                >
                   <MovieContainer onSubmit={this.handleCreate} />
                </PopUp>
                
                <PopUp
                  isOpen={showPopup && isSuccess}
                  onClose={this.handleClose}
                >
                   <Success /> // `SuccessModal` or `SuccessContainer`
                </PopUp>
    
```